components/Ink/src/private/MDCInkLayer.h

```diff
@@ -17,86 +17,98 @@
 #import <QuartzCore/QuartzCore.h>
 #import <UIKit/UIKit.h>
 
+@protocol MDCInkLayerDelegate;
+
 /**
  A Core Animation layer that draws and animates the ink effect.
 
  Quick summary of how the ink ripple works:
 
- 1. On touch down, blast initiates from the touch point.
- 2. On touch down hold, it continues to spread, but will gravitate to the center point
+ 1. On touch down, ink spreads from the touch point.
+ 2. On touch down hold, ink continues to spread, but will gravitate to the center point
     of the view.
- 3. On touch up, the ink ripple will lose energy, opacity will start to decrease.
+ 3. On touch up, the ink ripple opacity will start to decrease.
  */
-@interface MDCInkLayer : CALayer
+@interface MDCInkLayer : CAShapeLayer
 
-/** Clips the ripple to the bounds of the layer. */
-@property(nonatomic, assign, getter=isBounded) BOOL bounded;
+/**
+ Ink layer animation delegate. Clients set this delegate to receive updates when ink layer
+ animations start and end.
+ */
+@property(nonatomic, weak, nullable) id<MDCInkLayerDelegate> animationDelegate;
 
-/** Maximum radius of the ink. No maximum if radius is 0 or less. This value is ignored if
- @c bounded is set to |YES|.*/
-@property(nonatomic, assign) CGFloat maxRippleRadius;
+/**
+ The start ink ripple spread animation has started and is active.
+ */
+@property(nonatomic, assign) BOOL startAnimationActive;
```

Comment:
Is this meant to be settable? Consider making it readonly.

Reply:
Done. This is only changed within the class implementation.